### PR TITLE
Initial integration of primereact

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: common/src/main/scala/explore/components/primereact/EnumDropdownView.scala
Comment (cquiroz) [inline common/src/main/scala/explore/components/primereact/EnumDropdownView.scala:19]: Would these be eventually moved to lucuma-ui?

File: common/src/main/scala/explore/components/primereact/EnumDropdownView.scala
Comment (cquiroz) [inline common/src/main/scala/explore/components/primereact/EnumDropdownView.scala:11]: Could we use only the minimal imports from `react.common`?

File: common/src/main/webapp/less/style.less
Comment (cquiroz) [inline common/src/main/webapp/less/style.less:1075]: Should we switch to sass step by step?

**Review by cquiroz — COMMENTED** [review]
It is obviously hard to review 50k of code :D
Some questions:
* Are the theme css imported `as is`. Can we just replace them if they change upstream?
* Do the currently created css variables play a role?

As a general thing I'd like that we wait until we have the production  release stable before  merging this to avoid having two styles in production

**Comment by rpiaggio** [comment]
Thank you for tackling this!

Would it be possible to reference the prime-react files instead of copying them over? That way we don't have to manually keep them up to date.

File: common/src/main/scala/explore/components/primereact/EnumDropdownView.scala
Comment (toddburnside) [reply common/src/main/scala/explore/components/primereact/EnumDropdownView.scala:19]: Yes. That is the idea. It was just easiest to start with them here.

**Comment by toddburnside** [comment]
> * Are the theme css imported `as is`. Can we just replace them if they change upstream?
 
At the moment, they are imported `as is` from a modified instance of the theme designer (that I need to put in github as a private repository). We can just replace them if they change in that theme designer project, but I think we should find a better solution. Raul and I have discussed somehow pulling them out of a dependency jar or creating an npm package. I need to re-read the theme designer license (https://www.primefaces.org/designer/licenses) to see what we can do within the limits of the license.

**Comment by toddburnside** [comment]
> * Do the currently created css variables play a role?

Do you mean the css variables in the currently deployed explore (non-primereact)? If so, I think most (or all) of the variables related to color should probably be superseded eventually by the primereact CSS variables. We can tweak the theme colors if needed.

**Comment by toddburnside** [comment]
> As a general thing I'd like that we wait until we have the production release stable before merging this to avoid having two styles in production

Agreed. That's why this is a draft PR. :)

**Comment by toddburnside** [comment]
> Would it be possible to reference the prime-react files instead of copying them over? That way we don't have to manually keep them up to date.

Reference in what way?

**Comment by rpiaggio** [comment]
> Reference in what way?

Are they available in an npm package?

**Comment by toddburnside** [comment]
> Are they available in an npm package?

They aren't currently. I'm pretty sure that creating an npm package with the sass code (designer) would violate the license. But, it may be OK to create a package with the processed CSS file. I need to read the license again.

File: common/src/main/scala/explore/components/primereact/FormEnumDropdownView.scala
Comment (toddburnside) [inline common/src/main/scala/explore/components/primereact/FormEnumDropdownView.scala:21]: For this and the option drop down view for enums, the only reason to have a `Form` version and a `non-Form` version is that the `non-Form` version can return the builder, while the `Form` version can't because it also has a label. If we do a full facade or don't expose the builder, there is probably no real reason to have the `non-Form` variant.

File: common/src/main/scala/explore/components/primereact/FormInputText.scala
Comment (toddburnside) [inline common/src/main/scala/explore/components/primereact/FormInputText.scala:25]: Having the label here can reduce some code while using, but more important in my opinion is that it ensures that the id of the input matches the `htmlFor` of the label for accessibility reasons.

File: common/src/main/webapp/sass/explore-primereact.scss
Comment (rpiaggio) [inline common/src/main/webapp/sass/explore-primereact.scss:1]: We don't need to enclose the theme imports in `body`. It will save us some space since `body` is copied to every selector in the theme.

File: explore/src/main/webapp/main.jsx
Comment (rpiaggio) [inline explore/src/main/webapp/main.jsx:4]: Do you think we can avoid importing prime icons?

**Review by rpiaggio — APPROVED** [review]
LGTM

**Pull request merged**
